micov/_io.py

```diff
@@ -190,6 +210,39 @@ def parse_genome_lengths(lengths):
     return df[[genome_id_col, length_col]].rename(rename)
 
 
+def parse_taxonomy(taxonomy):
+    with open(taxonomy) as fp:
+        first_line = fp.readline()
+
+    has_header = _test_has_header_taxonomy(first_line)
+    df = pl.read_csv(taxonomy, separator='\t', has_header=has_header)
+    genome_id_col = df.columns[0]
+    taxonomy_col = df.columns[1]
+
+    genome_ids = df[genome_id_col]
+    if len(genome_ids) != len(set(genome_ids)):
+        raise ValueError(f"'{genome_id_col}' is not unique")
+
+    rename = {genome_id_col: COLUMN_GENOME_ID,
+              taxonomy_col: COLUMN_TAXONOMY}
+
+    return df[[genome_id_col, taxonomy_col]].rename(rename)
+
+
+def set_taxonomy_as_id(coverages, taxonomy):
+    missing = (set(coverages[COLUMN_GENOME_ID]) -
+               set(taxonomy[COLUMN_GENOME_ID]))
+    if len(missing) > 0:
+        raise ValueError(f"{len(missing)} genome(s) appear unrepresented in "
+                         f"the taxonomy information, examples: "
+                         f"{sorted(missing)[:5]}")
+
+    return (coverages
+            .join(taxonomy, on=COLUMN_GENOME_ID, how='left')
```

Comment:
why `left` and not `inner`?

Reply:
I used `left join` because i wanted to catch missing genome ids in taxonomy table. However, here left join and inner join should produce the same results because missing genome ids in the taxonomy table will be caught by these lines right before the join
```
missing = (set(coverages[COLUMN_GENOME_ID]) -
           set(taxonomy[COLUMN_GENOME_ID]))
if len(missing) > 0:
    raise ValueError(f"{len(missing)} genome(s) appear unrepresented in "
                     f"the taxonomy information, examples: "
                     f"{sorted(missing)[:5]}")
```

Updating this to inner join in the next commit.